### PR TITLE
[MRG+1] Simplify and clean multicolor_line example

### PR DESCRIPTION
**Pull request by dstansby**
I've taken the number of figures down to 1, which I think makes the whole example more understandable. Also a general clean and move of the example to the correct folder.

### CODE REVIEW

File: examples/pylab_examples/multicolored_line.py [hunk 1]
```diff
@@ -1,53 +0,0 @@
...
-from matplotlib.colors import ListedColormap, BoundaryNorm
-
-x = np.linspace(0, 3 * np.pi, 500)
-y = np.sin(x)
-z = np.cos(0.5 * (x[:-1] + x[1:]))  # first derivative
-
-# Create a colormap for red, green and blue and a norm to color
-# f' < -0.5 red, f' > 0.5 blue, and the rest green
-cmap = ListedColormap(['r', 'g', 'b'])
-norm = BoundaryNorm([-1, -0.5, 0.5, 1], cmap.N)
```
Comment (tacaswell) [inline examples/pylab_examples/multicolored_line.py]: It is a shame to loose the boundary norm example here.  Maybe do two axes, one with continuous, one with boundary norm?

Comment (dstansby) [inline examples/pylab_examples/multicolored_line.py]: :+1: That shouldn't be too hard to add back in

**Comment by dstansby** [comment]
figure_1

File: examples/lines_bars_and_markers/multicolored_line.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+from matplotlib.colors import ListedColormap, BoundaryNorm
+
+x = np.linspace(0, 3 * np.pi, 500)
+y = np.sin(x)
+dydx = np.cos(0.5 * (x[:-1] + x[1:]))  # first derivative
+
+# Create a set of line segments so that we can color them individually
+# This creates the points as a N x 1 x 2 array so that we can stack points
+# together easily to get the segments. The segments array for line collection
+# needs to be numlines x points per line x 2 (x and y)
```
Comment (anntzer) [inline examples/lines_bars_and_markers/multicolored_line.py]: `(number of lines) x (points per line) x 2`, perhaps.

**Comment by NelleV** [comment]
Thanks @dstansby ! This looks great!

**Pull request merged**